### ISSUE TITLE
Waterfox Classic 2019.12-classic, FreeBSD-CURRENT: Check spelling: enabled dictionaries are no longer effective; will not build …

### ISSUE THREAD

**Issue by grahamperrin** [MrAlex94/Waterfox#1298]
Screenshot from <https://github.com/MrAlex94/Waterfox/issues/1220#issuecomment-564363642>, Waterfox Classic 2019.12 (20191210201058): 

![image](https://user-images.githubusercontent.com/192271/70588973-56165480-1bc6-11ea-94ce-2aaa02f3ee4e.png)

Reproducible with a different profile in safe mode: 

```text
grahamperrin@momh167-gjp4-8570p:~ % date ; uname -v && waterfox --safe-mode -p test
Wed 11 Dec 2019 04:35:03 GMT
FreeBSD 13.0-CURRENT #45 r355536: Mon Dec  9 03:20:46 GMT 2019     root@momh167-gjp4-8570p:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG 
error: /usr/local/lib/waterfox/dictionaries/en-US.aff: cannot open
error: /usr/local/lib/waterfox/dictionaries/en-US.dic: cannot open
error: /usr/local/lib/waterfox/dictionaries/en-US.aff: cannot open
grahamperrin@momh167-gjp4-8570p:~ % 
```

Not reproducible with Waterfox Classic 2019.12 (20191206122057) on Linux. 

How might I troubleshoot this?

**Comment by hawkeye116477** [comment]
@grahamperrin Hasn't BSD hunspell? You can remove bundled dictionaries and make symlink to system's hunspell dictionaries.

**Comment by grahamperrin** [comment]
Thanks, similarly not working: 

![image](https://user-images.githubusercontent.com/192271/70955729-c5c68c80-2069-11ea-818a-6e4d222f4ae8.png)

```text
root@momh167-gjp4-8570p:/usr/local/lib/waterfox/dictionaries # ls -hlR
total 3
lrwxr-xr-x  1 root  wheel    39B Dec 17 00:55 en_GB-ise.aff -> /usr/local/share/hunspell/en_GB-ise.aff
lrwxr-xr-x  1 root  wheel    39B Dec 17 00:55 en_GB-ise.dic -> /usr/local/share/hunspell/en_GB-ise.dic
lrwxr-xr-x  1 root  wheel    35B Dec 17 00:56 en_GB.aff -> /usr/local/share/hunspell/en_GB.aff
lrwxr-xr-x  1 root  wheel    35B Dec 17 00:56 en_GB.dic -> /usr/local/share/hunspell/en_GB.dic
drwxr-xr-x  2 root  wheel     4B Dec 17 00:52 setaside

./setaside:
total 361
-rw-r--r--  1 root  wheel   3.0K Dec  5 18:18 en-US.aff
-rw-r--r--  1 root  wheel   563K Dec  5 18:18 en-US.dic
root@momh167-gjp4-8570p:/usr/local/lib/waterfox/dictionaries # 
```

**Comment by grahamperrin** [comment]
@jbeich any idea?

**Comment by jbeich** [comment]
@grahamperrin, try removing [files/patch-bug991253](https://github.com/freebsd/freebsd-ports/blob/319ab8ffcf01f38eb64a200d2275ebad205adc6d/www/waterfox/files/patch-bug991253) per https://github.com/freebsd/freebsd-ports/commit/68589eb6ce0b4d65a259fb5c22d754518e46c441

**Comment by grahamperrin** [comment]
Thanks, unfortunately (with a more recent ports tree) there's a compilation failure: 

```text
…
libxpcom_io.a.desc
rm -f libxpcom_io.a
/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/_virtualenv/bin/python /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/config/expandlibs_gen.py -o libxpcom_io.a.desc Unified_c_xpcom_io0.o nsLocalFileUnix.o Unified_cpp_xpcom_io0.o Unified_cpp_xpcom_io1.o  
gmake[5]: Leaving directory '/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/xpcom/io'
/usr/bin/c++ -std=gnu++11 -o Unified_cpp_xpcom_threads1.o -c -fvisibility=hidden -fvisibility-inlines-hidden -DNDEBUG=1 -DTRIMMED=1 -DOS_POSIX=1 -DOS_FREEBSD=1 -DOS_BSD=1 -DSTATIC_EXPORTABLE_JS_API -DMOZ_HAS_MOZGLUE -DMOZILLA_INTERNAL_API -DIMPL_LIBXUL -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/xpcom/threads -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/xpcom/threads -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/xpcom/build -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/caps -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/tools/profiler -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/ipc/ipdl/_ipdlheaders -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/ipc/chromium/src -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/ipc/glue -I/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/dist/include -I/usr/local/include/nspr -I/usr/local/include/nss -I/usr/local/include/nss/nss -I/usr/local/include -I/usr/local/include -I/usr/local/include/pixman-1  -fPIC  -DMOZILLA_CLIENT -include /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/mozilla-config.h -MD -MP -MF .deps/Unified_cpp_xpcom_threads1.o.pp -Qunused-arguments -DLIBICONV_PLUG -isystem /usr/local/include   -Qunused-arguments -Wall -Wc++11-compat -Wempty-body -Wignored-qualifiers -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wtype-limits -Wunreachable-code -Wwrite-strings -Wno-invalid-offsetof -Wclass-varargs -Wloop-analysis -Wc++11-compat-pedantic -Wc++14-compat -Wc++14-compat-pedantic -Wc++1z-compat -Wcomma -Wimplicit-fallthrough -Wstring-conversion -Wno-inline-new-delete -Wno-error=deprecated-declarations -Wno-error=array-bounds -Wformat -Wno-gnu-zero-variadic-macro-arguments -Wformat-security -Wno-unknown-warning-option -Wno-return-type-c-linkage -O2 -pipe -O3 -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing -DLIBICONV_PLUG -isystem /usr/local/include -fno-exceptions -fno-strict-aliasing -fno-rtti -ffunction-sections -fdata-sections -fno-exceptions -fno-math-errno -pipe -I/usr/local/include  -O2 -O3 -fno-omit-frame-pointer     /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/xpcom/threads/Unified_cpp_xpcom_threads1.cpp
error: build failed
gmake[5]: *** [/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/config/rules.mk:998: force-cargo-library-build] Error 101
gmake[5]: Leaving directory '/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/toolkit/library/rust'
gmake[4]: *** [/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/config/recurse.mk:73: toolkit/library/rust/target] Error 2
gmake[4]: *** Waiting for unfinished jobs....
3 warnings generated.
libxpcom_components.a.desc
rm -f libxpcom_components.a
/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/_virtualenv/bin/python /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/config/expandlibs_gen.py -o libxpcom_components.a.desc Unified_cpp_xpcom_components0.o  
gmake[5]: Leaving directory '/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/xpcom/components'
In file included from /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/xpcom/threads/Unified_cpp_xpcom_threads1.cpp:47:
In file included from /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/xpcom/threads/nsThread.cpp:24:
In file included from /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/dist/include/mozilla/CycleCollectedJSContext.h:16:
/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/dist/include/jsapi.h:1174:9: warning: field 'arrayProtoValues_' will be initialized after field 'forEachStatement_' [-Wreorder]
        arrayProtoValues_(true),
        ^
1 warning generated.
1 warning generated.
libxpcom_threads.a.desc
rm -f libxpcom_threads.a
/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/_virtualenv/bin/python /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/config/expandlibs_gen.py -o libxpcom_threads.a.desc Unified_cpp_xpcom_threads0.o Unified_cpp_xpcom_threads1.o  
gmake[5]: Leaving directory '/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build/xpcom/threads'
gmake[4]: Leaving directory '/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build'
gmake[3]: *** [/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/config/recurse.mk:33: compile] Error 2
gmake[3]: Leaving directory '/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build'
gmake[2]: *** [/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/Waterfox-2019.12-classic/config/rules.mk:452: all] Error 2
gmake[2]: Leaving directory '/usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox/work/.build'
===> Compilation failed unexpectedly.
Try to set MAKE_JOBS_UNSAFE=yes and rebuild before reporting the failure to
the maintainer.
*** Error code 1

Stop.
make[1]: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
*** Error code 1

Stop.
make: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
       34.99 real       101.88 user         7.01 sys
# echo $MAKE_JOBS_UNSAFE
yes
# 
```

**Comment by jbeich** [comment]
Downgrade lang/rust to 1.39 and try again.

**Comment by jbeich** [comment]
Also, the above snippet doesn't include actual error.

**Comment by grahamperrin** [comment]
Done, but then I hit the `devel/rust-cbindgen` dependency on `rust>=1.40.0`

Given the long list of `Makefile`s at <https://svnweb.freebsd.org/ports?view=revision&revision=520446>, might it be simplest to checkout the entire ports tree at a revision for an earlier date? 

`520295` maybe? https://svnweb.freebsd.org/ports?view=revision&revision=520295 (2019-12-17)

Or maybe from shortly before I began the build of what I currently use – build ID `20191210201058`. From my logs: 

> Tue Dec 10 19:18:35 GMT 2019

https://svnweb.freebsd.org/ports?view=revision&revision=519638 was early in the morning on the 10th …

**Comment by jbeich** [comment]
>Done, but then I hit the `devel/rust-cbindgen` dependency on `rust>=1.40.0`

`lang/rust` maintainer did so for simplicity. Just replace 1.40 with 1.39 in /usr/ports/Mk/Uses/cargo.mk. 

> might it be simplest to checkout the entire ports tree at a revision for an earlier date?

Yep but if you have other ports installed using newer tree it may cause conflicts.

**Comment by hawkeye116477** [comment]
You can also try patch from #1320.

**Comment by grahamperrin** [comment]
@hawkeye116477 thanks, that merged commit should simplify things for a future release. 

----

For now: below, I can't figure out _where_ Python `2.7` _was specified_. 

```
…
# make configure
===>   waterfox-2019.12.c depends on package: nspr>=4.16 - found
===>   waterfox-2019.12.c depends on package: nss>=3.32.1 - found
===>   waterfox-2019.12.c depends on package: icu>=59.1,1 - found
===>   waterfox-2019.12.c depends on package: libevent>=2.1.8 - found
===>   waterfox-2019.12.c depends on package: harfbuzz>=1.4.7 - found
===>   waterfox-2019.12.c depends on package: graphite2>=1.3.10 - found
===>   waterfox-2019.12.c depends on package: png>=1.6.31 - found
===>   waterfox-2019.12.c depends on package: libvorbis>=1.3.5,3 - found
===>   waterfox-2019.12.c depends on package: libvpx>=1.5.0 - found
===>   waterfox-2019.12.c depends on package: sqlite3>=3.19.3 - found
===>   waterfox-2019.12.c depends on package: py27-sqlite3>0 - found
===>   waterfox-2019.12.c depends on package: v4l_compat>0 - found
===>   waterfox-2019.12.c depends on executable: autoconf-2.13 - found
===>   waterfox-2019.12.c depends on executable: yasm - found
===>   waterfox-2019.12.c depends on executable: zip - found
===>   waterfox-2019.12.c depends on package: rust>=1.35 - not found
===>  rust-1.39.0 needs Python 3.3 at least, but 2.7 was specified.
*** Error code 1

Stop.
make[2]: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/lang/rust
*** Error code 1

Stop.
make[1]: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
*** Error code 1

Stop.
make: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
# 
…
# python -V
Python 3.7.6
# grep -i Python ./Makefile 
                ${PYTHON_PKGNAMEPREFIX}sqlite3>0:databases/py-sqlite3@${PY_FLAVOR} \
# grep -i Python ../../lang/rust/Makefile
USES=           pkgconfig python:3.3+,build ssl tar:xz
X_PY_CMD=       ${PYTHON_CMD} ${WRKSRC}/x.py
                -e 's,%PYTHON_CMD%,${PYTHON_CMD},' \
# 
```

**Comment by jbeich** [comment]
@grahamperrin, rebuild lang/rust separately to avoid [FreeBSD bug 233723](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=233723). Mk/bsd.gecko.mk defines `USES=python:2.7,build` but lang/rust defines `USES=python:3.3+,build`, both try to set `PYTHON_VERSION` variable.

**Comment by grahamperrin** [comment]
> try removing [files/patch-bug991253](https://github.com/freebsd/freebsd-ports/blob/319ab8ffcf01f38eb64a200d2275ebad205adc6d/www/waterfox/files/patch-bug991253) per https://github.com/freebsd/freebsd-ports/commit/68589eb6ce0b4d65a259fb5c22d754518e46c441

OK, I have a new build `20200103060435` but unfortunately, still the same symptoms as in the opening post. 

When checking is enabled, it's as if _nearly all_ words are mis-spelt.

**Comment by grahamperrin** [comment]
Note to self: running order

```text
sh
date ; uname -v
whoami
cd /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/
ls -hl
grep 2019.12 ./waterfox.revive.diff.txt
rm -r freebsd-ports
git clone https://github.com/freebsd/freebsd-ports
cd freebsd-ports
sed -i '' 's/1\.40/1.39/' Mk/Uses/cargo.mk
grep 1.39 Mk/Uses/cargo.mk
rm -r lang/rust
svn checkout svn://svn.freebsd.org/ports/head/lang/rust@518792 lang/rust
cd lang/rust
make configure
cd ../../..
cd freebsd-ports
git am $OLDPWD/waterfox.revive.diff.txt
rm www/waterfox/files/patch-bug991253
pkg remove rust
cd lang/rust
make install
cd ../../www/waterfox
make configure
make install
pkg lock waterfox
pkg lock rust
```

**Comment by grahamperrin** [comment]
### 2020.01-classic

tl;dr probably time to move on (fifteen months past the expiry date was longer than I expected) …

```text
root@momh167-gjp4-8570p:~ # sh
# date ; uname -v
Sat Jan 25 07:52:37 GMT 2020
FreeBSD 13.0-CURRENT #50 r356618: Sat Jan 11 04:50:38 GMT 2020     root@momh167-gjp4-8570p:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG 
# whoami
root
# cd /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/
# ls -hl
total 1587
-rw-r--r--   1 grahamperrin  grahamperrin    14K Jan  2 21:27 2020-01-02 21:25 make configure failed.txt
-rw-r--r--   1 grahamperrin  grahamperrin   7.6K Dec 11 03:10 Waterfox Classic 2019.12 (20191210201058) revive log summary.txt
-rw-r--r--   1 grahamperrin  grahamperrin   9.6M Jan  3 07:37 Waterfox Classic 2019.12 (20200103060435) revive log summary.txt
drwxr-xr-x  70 root          grahamperrin    85B Jan 25 07:33 freebsd-ports
-rw-r--r--   1 grahamperrin  grahamperrin   1.7M Jan 25 06:28 waterfox.revive.diff.txt
# grep 2020.01 ./waterfox.revive.diff.txt
+DISTVERSION=   2020.01-classic
+SHA256 (MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz) = 9dfe18230c4a936b51aa05eed5c7b85f32c4e552370c920499f12631616fc915
+SIZE (MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz) = 407507367
# grep 2450,6 -B 4 -A 8 waterfox.revive.diff.txt
diff --git a/www/Makefile b/www/Makefile
index 74e480f1cb0e..6ab335fbeede 100644
--- a/www/Makefile
+++ b/www/Makefile
@@ -2450,6 +2450,7 @@
     SUBDIR += w3mir
     SUBDIR += waccess
     SUBDIR += wadcomblog
+    SUBDIR += waterfox
     SUBDIR += webalizer
     SUBDIR += webbrowser
     SUBDIR += webcheck

# rm -r freebsd-ports
# git clone https://github.com/freebsd/freebsd-ports
Cloning into 'freebsd-ports'...
remote: Enumerating objects: 155, done.
remote: Counting objects: 100% (155/155), done.
remote: Compressing objects: 100% (123/123), done.
remote: Total 5146504 (delta 47), reused 99 (delta 31), pack-reused 5146349
Receiving objects: 100% (5146504/5146504), 1.65 GiB | 8.72 MiB/s, done.
Resolving deltas: 100% (2522870/2522870), done.
Updating files: 100% (138136/138136), done.
# cd freebsd-ports
# git am $OLDPWD/waterfox.revive.diff.txt
Applying: www/waterfox: temporarily revive
.git/rebase-apply/patch:879: trailing whitespace.
 
.git/rebase-apply/patch:912: trailing whitespace.
 
.git/rebase-apply/patch:913: trailing whitespace.
diff --git toolkit/mozapps/extensions/internal/XPIProvider.jsm toolkit/mozapps/extensions/internal/XPIProvider.jsm 
.git/rebase-apply/patch:937: trailing whitespace.
 
.git/rebase-apply/patch:943: trailing whitespace.
 
warning: squelched 3042 whitespace errors
warning: 3047 lines add whitespace errors.
# cd www/waterfox
# make configure
===>   waterfox-2020.01.c depends on file: /usr/local/sbin/pkg - found
=> MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz doesn't seem to exist in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/distfiles/.
=> Attempting to fetch https://codeload.github.com/MrAlex94/Waterfox/tar.gz/2020.01-classic?dummy=/MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz
fetch: https://codeload.github.com/MrAlex94/Waterfox/tar.gz/2020.01-classic?dummy=/MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz: size unknown
fetch: https://codeload.github.com/MrAlex94/Waterfox/tar.gz/2020.01-classic?dummy=/MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz: size of remote file is not known
MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz           388 MB 6286 kBps 01m03s
===> Fetching all distfiles required by waterfox-2020.01.c for building
===>  Extracting for waterfox-2020.01.c
=> SHA256 Checksum OK for MrAlex94-Waterfox-2020.01-classic_GH0.tar.gz.
===>  Patching for waterfox-2020.01.c
===>  Applying FreeBSD patches for waterfox-2020.01.c
No such line 7185 in input file, ignoring
No such line 326 in input file, ignoring
No such line 93 in input file, ignoring
No such line 878 in input file, ignoring
No such line 1265 in input file, ignoring
No such line 237 in input file, ignoring
No such line 230 in input file, ignoring
No such line 2422 in input file, ignoring
No such line 3942 in input file, ignoring
Ignoring previously applied (or reversed) patch.
1 out of 1 hunks ignored--saving rejects to netwerk/protocol/http/HttpChannelParent.cpp.rej
Ignoring previously applied (or reversed) patch.
1 out of 1 hunks ignored--saving rejects to netwerk/protocol/http/HttpChannelParent.h.rej
=> FreeBSD patch patch-bug1430508 failed to apply cleanly.
=> Patch(es)  patch-addon-search patch-browser-app-nsBrowserApp.cpp patch-bug1036008 patch-bug1134077 patch-bug1144632 patch-bug1186967 patch-bug1203273 patch-bug1231203 patch-bug1246883 patch-bug1275062 patch-bug1288587 patch-bug1321069 patch-bug1321960 patch-bug1324042 patch-bug1325888 patch-bug1331209 patch-bug1334465 patch-bug1334981 patch-bug1338764 patch-bug1340039 patch-bug1343147 patch-bug1345058 patch-bug1346126 patch-bug1346535 patch-bug1347984 patch-bug1352453 patch-bug1355576 patch-bug1356854 patch-bug1357523 patch-bug1360278 patch-bug1360603 patch-bug1363038 patch-bug1370508 patch-bug1370791 patch-bug1372107 patch-bug1373371 patch-bug1377158 patch-bug1377587 patch-bug1381761 patch-bug1381815 patch-bug1382359 patch-bug1383501 patch-bug1384701 patch-bug1385013 patch-bug1385609 patch-bug1387170 patch-bug1387811 patch-bug1388020 patch-bug1388319 patch-bug1388744 patch-bug1393627 patch-bug1394146 patch-bug1394601 patch-bug1395246 patch-bug1396722 patch-bug1397114 patch-bug1397671 patch-bug1397686 patch-bug1397795 patch-bug1397989 patch-bug1398021 patch-bug1398539 patch-bug1398581 patch-bug1398806 patch-bug1398983 patch-bug1399091 patch-bug1399177 patch-bug1399336 patch-bug1399412 patch-bug1399520 patch-bug1399939 patch-bug1400238 patch-bug1400540 patch-bug1400674 patch-bug1400839 patch-bug1401063 patch-bug1401416 patch-bug1401522 patch-bug1401573 patch-bug1401726 patch-bug1401876 patch-bug1401878 patch-bug1401909 patch-bug1401992 patch-bug1402025 patch-bug1402094 patch-bug1402205 patch-bug1402218 patch-bug1402366 patch-bug1402377 patch-bug1402380 patch-bug1402442 patch-bug1402469 patch-bug1402681 patch-bug1402715 patch-bug1402766 patch-bug1402871 patch-bug1402951 patch-bug1402966 patch-bug1402981 patch-bug1403282 patch-bug1403349 patch-bug1403465 patch-bug1403466 patch-bug1403473 patch-bug1403592 patch-bug1403721 patch-bug1403998 patch-bug1404057 patch-bug1404180 patch-bug1404323 patch-bug1404324 patch-bug1404568 patch-bug1404651 patch-bug1404714 patch-bug1404760 patch-bug1404910 patch-bug1405199 patch-bug1405267 patch-bug1405364 patch-bug1405377 patch-bug1405635 patch-bug1405655 patch-bug1405720 patch-bug1405761 patch-bug1405878 patch-bug1406222 patch-bug1406229 patch-bug1406254 patch-bug1406396 patch-bug1406474 patch-bug1406859 patch-bug1407001 patch-bug1407092 patch-bug1407243 patch-bug1407245 patch-bug1407414 patch-bug1407680 patch-bug1407936 patch-bug1408017 patch-bug1408276 patch-bug1408397 patch-bug1408631 patch-bug1408672 patch-bug1408990 patch-bug1409101 patch-bug1409277 patch-bug1409496 patch-bug1409649 patch-bug1409951 patch-bug1410134 patch-bug1410146 patch-bug1410420 patch-bug1410457 patch-bug1410634 patch-bug1410808 patch-bug1411415 patch-bug1411745 patch-bug1412011 patch-bug1412145 patch-bug1412420 patch-bug1412558 patch-bug1412741 patch-bug1412882 patch-bug1413143 patch-bug1413741 patch-bug1414292 patch-bug1414425 patch-bug1414452 patch-bug1414945 patch-bug1415291 patch-bug1415441 patch-bug1415582 patch-bug1415598 patch-bug1415883 patch-bug1415946 patch-bug1416307 patch-bug1416529 patch-bug1417661 patch-bug1417751 patch-bug1417797 patch-bug1417869 patch-bug1418009 patch-bug1418152 patch-bug1418757 patch-bug1418854 patch-bug1418894 patch-bug1419508 patch-bug1419762 patch-bug1420001 patch-bug1421009 patch-bug1421963 patch-bug1422036 patch-bug1422389 patch-bug1422518 patch-bug1422545 patch-bug1423647 patch-bug1423770 patch-bug1424311 patch-bug1424373 patch-bug1424469 patch-bug1425107 patch-bug1425267 patch-bug1425520 patch-bug1425612 patch-bug1425780 patch-bug1425878 patch-bug1426042 patch-bug1426087 patch-bug1426783 patch-bug1426801 patch-bug1426869 patch-bug1426996 patch-bug1427118 patch-bug1427126 patch-bug1427480 patch-bug1428589 patch-bug1428612 patch-bug1429373 patch-bug1429666 patch-bug1429764 applied cleanly.
*** Error code 1

Stop.
make[1]: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
*** Error code 1

Stop.
make: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
# mkdir files/setaside
# mv files/patch-bug1430508 files/setaside/
# mv files/patch-bug991253 files/setaside/
# make configure
===>  Patching for waterfox-2020.01.c
===>  Applying FreeBSD patches for waterfox-2020.01.c
Ignoring previously applied (or reversed) patch.
2 out of 2 hunks ignored--saving rejects to browser/app/profile/firefox.js.rej
Ignoring previously applied (or reversed) patch.
2 out of 2 hunks ignored--saving rejects to toolkit/mozapps/extensions/internal/AddonRepository.jsm.rej
No such line 7185 in input file, ignoring
Ignoring previously applied (or reversed) patch.
1 out of 1 hunks ignored--saving rejects to toolkit/mozapps/extensions/internal/XPIProvider.jsm.rej
=> FreeBSD patch patch-addon-search failed to apply cleanly.
*** Error code 1

Stop.
make[1]: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
*** Error code 1

Stop.
make: stopped in /usr/home/grahamperrin/Documents/IT/Waterfox/sandpit/freebsd-ports/www/waterfox
# 
```

– etc.; various patches fail to apply.

----

Re: https://github.com/MrAlex94/Waterfox/issues/1298#issuecomment-570142278 I'll try an earlier revision of the ports tree, if this doesn't work then I'll draw a line.

**Comment by grahamperrin** [comment]
> if this doesn't work then I'll draw a line.

No go, unfortunately. The ride was great whilst it lasted!

**Comment by ApeKattQuest-MonkeyPython** [comment]
hi, I just started using Waterfox classic (2021.08.1) and dictionaries do not work, I go to various firefox plugin to install and, while it lets me install and also says "remove" when accessing the page again https://addons.mozilla.org/en-GB/firefox/addon/norsk-bokm%C3%A5l-ordliste/ no dictionary shows up in text fields, it's stuck on US english
in plugins there is no section "for dictionaries" as the old firefox i had had.

is this related to this? is there anything I can do to try to make it works?
